### PR TITLE
Initial commit of the module service_container_permissions.

### PR DESCRIPTION
**Pull request by drupol**

### CODE REVIEW

File: modules/service_container_permissions/service_container_permissions.module [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * Main module file for the service_container_permissions module.
+ */
+
+// -----------------------------------------------------------------------
+// Contrib Hooks
+
+/**
+ * Implements hook_permission().
+ */
+function service_container_libraries_permission() {
```
Comment (fabianx-ai) [inline modules/service_container_permissions/service_container_permissions.module]: service_container_libraries_permission

**Comment by fabianx-ai** [comment]
There is more work needed:

https://www.drupal.org/node/2311427

**Comment by drupol** [comment]
Of course this module and service_container_libraries are just initial version, there are work to be done.

**Comment by dawehner** [comment]
- it think we should cache that information, because otherwise the permissions pages and the modules pages will get slower

**Comment by dawehner** [comment]
note: hook_permission itself is not cached in D7

**Comment by drupol** [comment]
@dawehner I have to clear cache to get into hook_permission(), it means that it's cached no ?

File: modules/service_container_permissions/service_container_permissions.module [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+          }
+          unset($permissions['permission_callbacks'][$index]);
+        }
+      }
+      $all_permissions = array_merge_recursive($all_permissions, $permissions);
+    }
+  }
+  unset($all_permissions['permission_callbacks']);
+
+  return $all_permissions;
```
Comment (fabianx-ai) [inline modules/service_container_permissions/service_container_permissions.module:50]: Daniel is right that we should use some kind of caching for this. If not then admin pages that need all permissions could potentially get very slow ...

Comment (drupol) [inline modules/service_container_permissions/service_container_permissions.module:50]: cache_set/cache_get then ?

**Pull request closed without merging**